core/src/main/java/org/apache/carbondata/core/scan/partition/PartitionUtil.java

```diff
@@ -71,6 +72,10 @@ public static Object getDataBasedOnDataType(String data, DataType actualDataType
         default:
           return data;
       }
+    } catch (NumberFormatException ex) {
+      return new IllegalArgumentException(ex.getMessage());
```

Comment:
can throw MalformedCarbonException instead of runtime.
Its not good coding practice to return the exception object instead of throwing it